content/docs/user-guide/scripting/script-canvas/editor-reference/text-replacement.md

```diff
@@ -0,0 +1,499 @@
+
+ ![Image 0](/images/user-guide/scripting/script-canvas/text-replacement-0.png)
+
+Right click on any of the nodes under AWSGameListPlayer
+
+![Image 1](/images/user-guide/scripting/script-canvas/text-replacement-1.png)
+
+And select "Generate Translation"
+
+This will open a file explorer window to an auto-generated file called: *AWSGameLiftPlayer.names*
+
+If you do not wish to make any changes to the auto-generated data, you can close the Script Canvas editor window and reopen it and you will see some changes have taken effect.
+
+![Image 2](/images/user-guide/scripting/script-canvas/text-replacement-2.png)
+
+All the names are now separated by a space and the naming follows the Script Canvas naming convention to use Camel Case.
+
+### Changing the node category
+
+We have succesfully generated translation data for a node, but its category is still under "Other". To change this we will now edit the generated file *AWSGameLiftPlayer.names*
+
+To change the category, navigate to the outermost "details" object and add:
+"category": "AWS Game Lift"
+
+In addition, the name can be simplified to only "Player" to avoid the redundancy.
+
+```
+{
+    "entries": [
+        {
+            "base": "AWSGameLiftPlayer",
+            "context": "BehaviorClass",
+            "variant": "",
+            "details": {
+                "category": "AWS Game Lift",
+                "name": "Player"
+            },
+            "methods": [
+                ... STRIPPED FOR SIZE ... (See Appendix 1 for complete file before/after)
+            ]
+        }
+    ]
+}
+```
+After saving the changes to *AWSGameLiftPlayer.names*, close the Script Canvas editor and open it again.
+
+The nodes are now in the proper category with the given name:
+
+![Image 3](/images/user-guide/scripting/script-canvas/text-replacement-3.png)
+
+
+## Example: Updating translation files
+
+In this case, we do not need to generate a new translation file, instead we just need to edit it.
+
+![Image 2](/images/user-guide/scripting/script-canvas/text-replacement-1.png)
+
+Select _"Explore Translation Data"_ it will open a file explorer on the file that contains the relevant translation data, in this case, *AWSGameLiftPlayer.names*.
+
+Open the file in your preferred text editor.
+
+Find and update any desired elements. 
+
+## Best Practices
+
+* Do not modify the `base`, `context`, `variant` or `typeid` fields
+* Use `Camel Case` syntax with spaces
+* Add the `tooltip` field to provide details and context
+* Use the `category` field to organize nodes
+* Review your changes by closing and reopening the Script Canvas editor
+* Watch for Asset Processor errors on .names files, ensure JSON syntax is correct
+* Commit .names files to source control, this will ensure they are tracked and changes are not overriden
+* Generating translation data will overwrite existing translation data, be careful not to lose work, when in doubt, use _Explore Translation Data_ first
```

Comment:
If you use Generate Translation when there is already a .names file for the object, does it warn you about the overwrite and give you a chance to cancel? If not, I think it should!